### DISCUSSION TITLE
How to stop all services running from a dir and its subdirs, and also kill all processes running from a dir and its subdirs

### DISCUSSION THREAD

**Discussion by dlhtox** [PowerShell/PowerShell#20580] [General]
I am looking to stop all services running from a particular directory and its subdirectories and also terminate all active processes after that that are running from that same directory. 

One PowerShell snippet for the services and then one for the processes.

Here is what I have so far for the processes, but it only does it for one dir, not all its subdirectories:

```
$files = gci "C:\Path\To\Files" -Filter "*.exe"

foreach($file in $files){
    Get-Process | 
    Where-Object {$_.Path -eq $file.FullName} | 
    Stop-Process -WhatIf
}
```

**Comment by MartinGC94** [comment]
You could do it like this: `Get-Process | Where-Object -Property Path -Like "C:\Program Files\Google\*" | Stop-Process`
